### PR TITLE
create an OSA from an OSW

### PR DESCRIPTION
**Pull request by brianlball**
right now, the only code we have to make OSAs programmatically is the JS in PAT. 
This will bring OSA creation to the SDK and in Notebooks.
This updates the older spreadsheet era methods to be compliant with the latest OSA schema.
In some instances, new methods had to be created to avoid breaking changes.

### CODE REVIEW

**Comment by brianlball** [comment]
https://github.com/NREL/OpenStudio-analysis-gem/issues/54

File: lib/openstudio/analysis/version.rb [hunk 1]
```diff
@@ -37,6 +37,6 @@ module OpenStudio
   module Analysis
     # format should be ^.*\-{1}[a-z]+[0-9]+
     # for example: -rc1, -beta6, -customusecase0
-    VERSION = '1.2.0'.freeze
+    VERSION = '1.3.0-0'.freeze
```
Comment (nllong) [inline lib/openstudio/analysis/version.rb:40]: release candidate?

```suggestion
    VERSION = '1.3.0-rc1'.freeze
```

File: spec/files/osw_project/measures/AddMonthlyJSONUtilityData/LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+OpenStudio(R), Copyright (c) 2008-2022, Alliance for Sustainable Energy, LLC. All rights reserved.
```
Comment (nllong) [inline spec/files/osw_project/measures/AddMonthlyJSONUtilityData/LICENSE.md:1]:
```suggestion
OpenStudio(R), Copyright (c) 2008-2023, Alliance for Sustainable Energy, LLC. All rights reserved.
```

File: spec/files/osw_project/measures/CalibrationReportsEnhanced/README.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+**Units:** ,
+**Required:** true,
+**Model Dependent:** false
+
+
+
+
+
+## Outputs
+
```
Comment (nllong) [inline spec/files/osw_project/measures/CalibrationReportsEnhanced/README.md:75]: maybe trim the number of lines here?

File: spec/files/osw_project/measures/GeneralCalibrationMeasurePercentChange/LICENSE.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+OpenStudio(R), Copyright (c) 2008-2022, Alliance for Sustainable Energy, LLC. All rights reserved.
```
Comment (nllong) [inline spec/files/osw_project/measures/GeneralCalibrationMeasurePercentChange/LICENSE.md:1]:
```suggestion
OpenStudio(R), Copyright (c) 2008-2023, Alliance for Sustainable Energy, LLC. All rights reserved.
```

File: spec/openstudio/excel_spec.rb [hunk 5]
```diff
@@ -288,7 +288,7 @@
     it 'should create a valid hash' do
       h = @excel.analysis
 
-      expect(h.analysis_type).to eq('nsga')
+      expect(h.analysis_type).to eq('nsga_nrel')
```
Comment (nllong) [inline spec/openstudio/excel_spec.rb:291]: nice update!

File: spec/openstudio/workflow_to_osa_spec.rb [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'spec_helper'
+require 'json-schema'
+
+describe 'Convert_an_OSW_to_OSA' do
```
Comment (nllong) [inline spec/openstudio/workflow_to_osa_spec.rb:39]: nice, tests!

**Review by nllong — APPROVED** [review]
amazing work!

File: lib/openstudio/analysis/version.rb (hunk 1 above)
Comment (brianlball) [reply lib/openstudio/analysis/version.rb:40]: yeah, rc1 is prob better.  Tests using this branch in Server pass  https://github.com/NREL/OpenStudio-server/tree/analysis-gem-1-3-0-0

**Pull request merged**